lib/kafo/scenario_manager.rb

```diff
@@ -98,23 +116,50 @@ def scenario_from_args
       end
     end
 
+    def show_scenario_diff(prev_scenario, new_scenario)
+      say ::HighLine.color("Scenarios are being compared, that may take a while...", :cyan)
+      prev_conf = load_and_setup_configuration(prev_scenario)
+      new_conf = load_and_setup_configuration(new_scenario)
+      print_scenario_diff(prev_conf, new_conf)
+    end
+
     def confirm_scenario_change(new_scenario)
-      if (ARGV & ['--interactive', '-i']).any?
-        # TODO: show option diff
-        wizard = KafoWizards.wizard(:cli, 'Confirm installation scenario selection',
-          :description => "You are trying to replace existing installation with different scenario. This may lead to unpredictable states. Please confirm that you want to proceed.")
-        wizard.entries << wizard.factory.button(:proceed, :label => 'Proceed with selected installation scenario', :default => false)
-        wizard.entries << wizard.factory.button(:cancel, :label => 'Cancel Installation', :default => true)
-        result = wizard.run
-        if result == :cancel
-          say 'Installation was cancelled by user'
-          KafoConfigure.exit(0)
+      unless ARGV.include?('--force')
+        if (ARGV & ['--interactive', '-i']).any?
+          show_scenario_diff(@previous_scenario, new_scenario)
+
+          wizard = KafoWizards.wizard(:cli, 'Confirm installation scenario selection',
+            :description => "You are trying to replace existing installation with different scenario. This may lead to unpredictable states. Please confirm that you want to proceed.")
+          wizard.entries << wizard.factory.button(:proceed, :label => 'Proceed with selected installation scenario', :default => false)
+          wizard.entries << wizard.factory.button(:cancel, :label => 'Cancel Installation', :default => true)
+          result = wizard.run
+          if result == :cancel
+            say 'Installation was cancelled by user'
+            KafoConfigure.exit(0)
+          end
+        else
+          message = "You are trying to replace existing installation with different scenario. This may lead to unpredictable states. " +
+          "Use --force to override. You can use --compare-scenarios to see the differences"
+          fail_now(message, :scenario_error)
         end
+      end
+    end
+
+    def print_scenario_diff(prev_conf, new_conf)
+      missing = new_conf.params_missing(prev_conf)
+      changed = new_conf.params_changed(prev_conf)
+
+      say ::HighLine.color("\nValues from previous installation that will be added by installer:", :yellow)
```

Comment:
please avoid adding \n as part of HighLine styling, it would cause problems if we set some style to use background color